File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) return 0;
        Set<Integer> numbers = new HashSet<>();
        int max = 1;

        for (int num: nums) {
            numbers.add(num);
        }       

        for (int curr: numbers) {
            int current = 1;
            if (numbers.contains(curr - 1)) continue;
            while(numbers.contains(curr + current)) {
                current++;
            }
            if (current > max) {
                max = current;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0;
        int r = nums.size() - 1;
        int minimum = nums[r];

        while (l <= r) {
            int middle = l + ((r - l) / 2);

            if (nums[middle] < minimum) {
                minimum = nums[middle];
                r = middle - 1;
            } else {
                l = middle + 1;
            }
        }

        return minimum;


    }
};
